src/components/Layout/Layout.tsx

```diff
@@ -32,7 +32,10 @@ export class Layout extends React.Component<LayoutProps, undefined> {
                 <div id="content">
                     <Grid/>
                     <Header/>
-                    <SideBar/>
+                    <SideBar history={this.props.history}>
```

Comment:
It's better to use Context to pass history to SideBar (we have history context in any element in Router), see Link component defenition